sonar-plugin-api/src/main/java/org/sonar/api/measures/AverageFormula.java

```diff
@@ -82,56 +63,10 @@ public AverageFormula setFallbackForMainMetric(Metric fallbackMetric) {
    */
   @Override
   public Measure calculate(FormulaData data, FormulaContext context) {
-    if (!shouldDecorateResource(data, context)) {
-      return null;
-    }
-
-    Measure result;
-    if (ResourceUtils.isFile(context.getResource())) {
-      result = calculateForFile(data, context);
-    } else {
-      result = calculateOnChildren(data, context);
-    }
-    return result;
-  }
-
-  private Measure calculateOnChildren(FormulaData data, FormulaContext context) {
-    Measure result = null;
-
-    double totalByMeasure = 0;
-    double totalMainMeasure = 0;
-    boolean hasApplicableChildren = false;
-
-    for (FormulaData childrenData : data.getChildren()) {
-      Double fallbackMeasure = fallbackMetric != null ? MeasureUtils.getValue(childrenData.getMeasure(fallbackMetric), null) : null;
-      Double childrenByMeasure = MeasureUtils.getValue(childrenData.getMeasure(byMetric), null);
-      Double childrenMainMeasure = MeasureUtils.getValue(childrenData.getMeasure(mainMetric), fallbackMeasure);
-      if (childrenMainMeasure != null && childrenByMeasure != null && childrenByMeasure > 0.0) {
-        totalByMeasure += childrenByMeasure;
-        totalMainMeasure += childrenMainMeasure;
-        hasApplicableChildren = true;
-      }
-    }
-    if (hasApplicableChildren) {
-      result = new Measure(context.getTargetMetric(), totalMainMeasure / totalByMeasure);
-    }
-    return result;
-  }
-
-  private Measure calculateForFile(FormulaData data, FormulaContext context) {
-    Measure result = null;
-
-    Double fallbackMeasure = fallbackMetric != null ? MeasureUtils.getValue(data.getMeasure(fallbackMetric), null) : null;
-    Double byMeasure = MeasureUtils.getValue(data.getMeasure(byMetric), null);
-    Double mainMeasure = MeasureUtils.getValue(data.getMeasure(mainMetric), fallbackMeasure);
-    if (mainMeasure != null && byMeasure != null && byMeasure > 0.0) {
-      result = new Measure(context.getTargetMetric(), mainMeasure / byMeasure);
-    }
-
-    return result;
+    throw fail();
   }
 
-  private static boolean shouldDecorateResource(FormulaData data, FormulaContext context) {
-    return !MeasureUtils.hasValue(data.getMeasure(context.getTargetMetric()));
+  private static RuntimeException fail() {
+    throw new UnsupportedOperationException("Unsupported since version 5.2. Decorators and formulas are not used anymore for aggregation measures. Please use org.sonar.api.ce.measure.MeasureComputer.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 202 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)